node/src/proxy_server/mod.rs

```diff
@@ -278,23 +279,24 @@ impl ProxyServer {
                 })
                 .clone()
         };
-        let host_name = match &return_route_info.server_name {
+        let host_name = match &return_route_info.server_name_opt {
             Some(name) => name.clone(),
-            None => "<unspecified server>".to_string(),
+            None => UNSPECIFIED_SERVER.to_string(),
```

Comment:
Regardless the outcome of the bigger scale issue below I also think this might be a mistake...
I think it doesn't make sense to make up our own string just to fill a void. 
Isn't that exactly why `Option` is a handy tool? 

Instead of:
```
 if host_name.ne(UNSPECIFIED_SERVER) {
                    self.subs
                        .as_ref()
                        ...
```                        
you can write better...
```
if let Some(host_name) = host_name_opt {
                    self.subs
                        .as_ref()
```                        
and it automatically cause that you won't need to care about the case if it's `None`. Literally, we ignore that case.

Reply:
The variable host_name, is used again in the same scope. It's more optimal according to me.

Reply:
Oh. You might mean the error!() log at the very end of this function. Well, it still looks better to me to work as long as possible with Option which allows you to avoid the always clumsy situation when you need to fill in a void just to satisfy the compiler...
That happens also here, you are forced to create a string saying UNSPECIFIED_SERVER even though, it's unlikely you will have to use it. Then developers coming to this place may ask themselves what is the purpose of having the two words.
They will eventually figure it out but you can do better if you don't bother them with it at all because it has a simple solution.
Just use host_name_opt, most of time it will fit best. When it comes to the error!() macro, notice that it perfectly sticks with what you need just write something like `host_name_opt.unwrap_or("<unspecified server>".to_string())` inside the macro (yes, as its argument). This way you will spare people from thinking too much about the mystical string created up there. (To me it looks like it was already a suboptimal solution from the person writing this code before you, so the blame doesn't fall on you).  

Hoping that you are okay with this breakdown. :pray:

Reply:
Your argument over here is strong enough to convince me.